includes/Plugin.php

```diff
@@ -140,6 +140,36 @@ public function __construct() {
 		add_action( 'action_scheduler_init', array( $this, 'schedule_token_migration_job' ) );
 		add_action( 'wc_square_init_payment_token_migration_v2', array( $this, 'register_payment_tokens_migration_scheduler' ) );
 		add_action( 'wc_square_init_payment_token_migration', '__return_false' );
+
+		add_filter( 'woocommerce_order_class', array( $this, 'order_class' ), 20 );
```

Comment:
Is there a better location for this and the associated method?

Reply:
Not sure if you saw the previous PR but we originally took a similar approach to solve this in #20. I haven't compared this line by line but I think they are pretty similar. We ended up closing that out (and basically leaving the deprecation notices) due to the feedback we received. Can you can a look at that PR and the comments there to see if the approach here is different enough to try again on?

Reply:
@dkotter I'll take a look, I went looking yesterday but couldn't find it.

Reply:
I've reviewed James's comments on the original PR and the upstream tickets.

Given the concerns about a race to the bottom, I think we're blocked until it's addressed upstream. Probably in `WC_Data` with either magic methods or a setter/getter for unsaved meta data.